src/webview/leetCodePreviewProvider.ts

```diff
@@ -11,19 +11,40 @@ class LeetCodePreviewProvider extends LeetCodeWebview {
 
     private node: IProblem;
     private description: IDescription;
+    private sideMode: boolean = false;
 
-    public async show(node: IProblem): Promise<void> {
+    public isSideMode(): boolean {
+        return this.sideMode;
+    }
+
+    public async show(node: IProblem, isSideMode: boolean = false): Promise<void> {
         this.description = this.parseDescription(await leetCodeExecutor.getDescription(node), node);
         this.node = node;
+        this.sideMode = isSideMode;
         this.showWebviewInternal();
+        if (this.sideMode) {
+            // Hide the side bar for better view area
+            commands.executeCommand("workbench.action.focusSideBar").then(() => {
```

Comment:
`await` here?

Reply:
I think `WebviewProvider.show` should not be an async function in the first place: every operation that needs to await something should be did before calling `show` ---- what is passed to `show` should be the resources prepared. 

The commands are also operations that don't need to be waited, just like `this.panel.reveal` does not wait for the panel to be revealed.

So I pushed a commit to move `getDescription` to `commands/show.ts/previewProblem`, and make `show` not an async function...